resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/ExecutorTask.scala

```diff
@@ -81,6 +83,14 @@ private[spark] object ExecutorTask
     // The driver will lookup the actual log URLs
     task.addEnv("SPARK_LOG_URL_" + LOG_KEY_FOR_ALLOC_ID, "${NOMAD_ALLOC_ID}")
 
+    if (conf.getOption("spark.jars").nonEmpty) {
+      task.addEnv("SPARK_EXECUTOR_CLASSPATH", conf.getOption("spark.jars").get
+        .split(",").map(jarLocation => new URI(jarLocation).getPath).mkString(":"))
```

Comment:
Is it ok that this will throw an exception if any `jarLocation` is not a valid URI?

Reply:
As Adam suggested the avoid handing the property `spark.jars` for manipulating the classpath (-cp), i removed this snippet of code completely.